classes/oai/ojs/OAIDAO.inc.php

```diff
@@ -172,20 +173,30 @@ public function getSetJournalSectionId($journalSpec, $sectionSpec, $restrictJour
         }
 
         $journalId = $journal->getId();
-        $sectionId = null;
+        $sectionId = 0;
 
         if (isset($sectionSpec)) {
-            $section = $this->sectionDao->getByAbbrev($sectionSpec, $journal->getId());
-            if (isset($section)) {
-                $sectionId = $section->getId();
-            } else {
-                $sectionId = 0;
+            $sectionIterator = $this->sectionDao->getByJournalId($journalId);
+
+            while ($section = $sectionIterator->next()) {
+                if ($sectionSpec == $this->toValidSetSpec($section->getLocalizedAbbrev())) {
```

Comment:
Was this meant to be a call to `OAIUtils::toValidSetSpec`?

Reply:
Oops, missed this one when moving the method